addons/im_livechat/models/discuss_channel.py

```diff
@@ -671,3 +671,40 @@ def _action_unfollow(self, partner=None, guest=None, post_leave_message=True):
             # sudo: discuss.channel - last operator left the conversation, state must be updated.
             channel_sudo.livechat_end_dt = fields.Datetime.now()
             Store(bus_channel=self).add(channel_sudo, "livechat_end_dt").bus_send()
+
+    def _forward_human_agent(self, agent, create_member_params=None, inviting_partner=None):
+        """Forward the conversation, initially handled by a bot, to a human agent.
+
+        :param agent: The user to forward the conversation to.
+        :param create_member_params: A dictionary of values to pass to the agent channel member
+            creation function.
+        :param inviting_partner: The partner that invited the agent to the channel. This will be
+            forwarded to the `_add_members` function for the "invite" notification.
+        """
+        self._add_members(
+            users=agent,
+            create_member_params={"livechat_member_type": "agent", **create_member_params},
+            inviting_partner=inviting_partner,
+        )
+        if bot_member := self.channel_member_ids.filtered(
+            lambda m: m.livechat_member_type == "bot"
+        ):
+            self._action_unfollow(partner=bot_member.partner_id, post_leave_message=False)
+        self.write(
+            {
+                "livechat_failure": "no_answer",
+                "livechat_operator_id": agent.partner_id,
+                "name": " ".join(
+                    [
+                        self.env.user.display_name
+                        if not self.env.user._is_public()
+                        else self.self_member_id.guest_id.name,
+                        agent.livechat_username
+                        if agent.livechat_username
+                        else agent.name,
+                    ],
+                ),
+            },
+        )
+        self._broadcast(agent.partner_id.ids)
```

Comment:
Outdated code?
This should already be part of `_add_members`: `member._bus_send("discuss.channel/joined", payload)`.
This whole `_broadcast` method should probably be removed in another PR, and replaced by appropriate `_add_members` or other relevant methods depending on the flow.